sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -347,44 +347,81 @@ object GpuCast extends Arm {
 
       // ansi cast from larger-than-integer integral-like types, to integer
       case (t @(LongType | _: DecimalType), IntegerType) if ansiMode =>
-        assertValuesInRange(input, GpuScalar.from(Int.MinValue, t),
-          GpuScalar.from(Int.MaxValue, t))
+        t match {
+          case LongType =>
+            assertValuesInRange[Long](input, Int.MinValue.toLong, Int.MaxValue.toLong)
+          case _ =>
+            assertValuesInRange[BigDecimal](input, BigDecimal(Int.MinValue),
+              BigDecimal(Int.MaxValue))
+        }
```

Comment:
Humnn, I'm not sure if this case is just `IntegerType` and `LongType` instead of `LongType` and `BigDecimal`? @NVnavkumar?

Reply:
Hi @ttnghia, are you referring to the case matching not being exhaustive? Thanks!

Reply:
Okay, so here the default case is `DecimalType`,  not `BigDecimal`. You may construct the bounds from the type `t`.